include/queue.hpp

```diff
@@ -1,42 +1,96 @@
+// This is a personal academic project. Dear PVS-Studio, please check it.
+// PVS-Studio Static Code Analyzer for C, C++, C#, and Java: http://www.viva64.com
+
 #ifndef MY_QUEUE_QUEUE_HPP
 #define MY_QUEUE_QUEUE_HPP
 
+#include <mutex>
+#include <condition_variable>
+
 template<typename T>
-struct node_t {
-    T value;
-    node_t* next;
+struct Node {
+    T m_value;
+    Node* m_next_ptr;
+    explicit Node(T value) : m_value{value}, m_next_ptr{nullptr} {};
 };
 
 template<typename T>
-class queue_t {
+class Queue {
 public:
-    queue_t() = default;
-    void enqueue(T value);
+    Queue() : m_size{0}, m_head_ptr{nullptr}, m_tail_ptr{nullptr} {};
+    ~Queue();
+    Queue(const Queue&) = delete;
+    Queue& operator=(const Queue&) = delete;
+    void enqueue(const T& value);
     T dequeue();
+    bool try_dequeue(T* value_ptr);
```

Comment:
I would prefer `std::optional<T> try_dequeue();` instead, but it is up to you whether to use argument as output